File: leetcode-python-go/1764. 通过连接另一个数组的子数组得到一个数组/main.go
package main

import (
	"fmt"
)

func canChoose(groups [][]int, nums []int) bool {
next:
	for _, g := range groups {
		for len(nums) >= len(g) {
			if equal(nums[:len(g)], g) {
				nums = nums[len(g):]
				continue next
			}
			nums = nums[1:]
		}
		return false
	}
	return true
}

func equal(a, b []int) bool {
	for i, x := range a {
		if x != b[i] {
			return false
		}
	}
	return true
}

func main() {
	groups := [][]int{{1, -1, -1}, {3, -2, 0}}
	nums := []int{1, -1, 0, 1, -1, -1, 3, -2, 0}
	results := canChoose(groups, nums)
	fmt.Println(results)
}

File: leetcode-python-go/1764. 通过连接另一个数组的子数组得到一个数组/main.py
from typing import List

groups = [[1,-1,-1],[3,-2,0]]
nums = [1,-1,0,1,-1,-1,3,-2,0]

def canChoose(groups: List[List[int]], nums: List[int]) -> bool:
    k = 0
    for g in groups:
        while k + len(g) <= len(nums):
            if nums[k: k + len(g)] == g:
                k += len(g)
                break
            k += 1
        else:
            return False
    return True

print(canChoose(groups, nums))